### PR TITLE
Split out fluent builder generators and clean up

### PR DESCRIPTION
**Pull request by jdisanti**
It looks like I will need to render specialized fluent builders for waiters. This PR is some prerequisite work to split fluent builder generation out of the fluent client generator into its own thing. This makes it possible to refactor the fluent builder generator to be more configurable in a follow up PR so that it can generate waiter fluent builders as well.

----

_By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice._

### CODE REVIEW

File: codegen-client/src/main/kotlin/software/amazon/smithy/rust/codegen/client/smithy/generators/client/FluentBuilderGenerator.kt [hunk 1]
```diff
@@ -69,45 +73,24 @@ class FluentBuilderGenerator(
...
-                > {
-                    let mut fluent_builder = client.$fnName();
-                    fluent_builder.inner = self;
-                    fluent_builder.send().await
-                }
-                """,
-                *scope,
-            )
-        }
+    private fun RustWriter.renderStruct() {
```
Comment (Velfi) [inline codegen-client/src/main/kotlin/software/amazon/smithy/rust/codegen/client/smithy/generators/client/FluentBuilderGenerator.kt:80]: I wish you could create private extensions in Rust.

File: codegen-client/src/main/kotlin/software/amazon/smithy/rust/codegen/client/smithy/generators/client/FluentBuilderGenerator.kt [hunk 2]
```diff
@@ -0,0 +1,370 @@
...
+                        config_override: #{None},
+                    }
+                }
+                """,
+                *scope,
+            )
+
+            rustTemplate(
+                """
+                /// Access the ${operationType.name} as a reference.\n
```
Comment (Velfi) [inline codegen-client/src/main/kotlin/software/amazon/smithy/rust/codegen/client/smithy/generators/client/FluentBuilderGenerator.kt:121]:
```suggestion
                /// Access the ${operationType.name} as a reference.
```

Comment (Velfi) [reply codegen-client/src/main/kotlin/software/amazon/smithy/rust/codegen/client/smithy/generators/client/FluentBuilderGenerator.kt:121]: Is this newline doing anything?

**Review by Velfi — APPROVED** [review]

File: codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/rustlang/RustWriter.kt [hunk 3]
```diff
@@ -385,6 +385,8 @@ fun <T : AbstractCodeWriter<T>> T.docs(
     setNewlinePrefix(newlinePrefix)
     val cleaned =
         text.lines()
+            // Remove leading empty line (frequently occurs with Kotlin block quotes)
+            .filterIndexed { index, line -> index != 0 || line.isNotBlank() }
```
Comment (jdisanti) [inline codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/rustlang/RustWriter.kt:389]:
```suggestion
```
This changes a lot more than I anticipated.

File: codegen-client/src/main/kotlin/software/amazon/smithy/rust/codegen/client/smithy/generators/client/FluentClientDecorator.kt [hunk 4]
```diff
@@ -22,6 +22,15 @@ import software.amazon.smithy.rust.codegen.core.smithy.generators.LibRsCustomiza
 import software.amazon.smithy.rust.codegen.core.smithy.generators.LibRsSection
 import software.amazon.smithy.rust.codegen.core.util.serviceNameOrDefault
 
+/**
+ * Generates the client via codegen decorator.
+ *
```
Comment (ysaito1001) [inline codegen-client/src/main/kotlin/software/amazon/smithy/rust/codegen/client/smithy/generators/client/FluentClientDecorator.kt:27]: Thanks for explaining the rationale!

**Pull request merged**